test/SILOptimizer/diagnostic_constant_propagation_floats.swift

```diff
@@ -0,0 +1,178 @@
+// RUN: %target-swift-frontend -emit-sil -primary-file %s -o /dev/null -verify
+//
+// These are tests for diagnostics produced by constant propagation pass
+// on floating-point operations
+
+import StdlibUnittest
+
+func testFloatConvertOverflow() {
+  let f1: Float = 1E38
+  _blackHole(f1)
+  let f2: Float = 1E39 // expected-warning {{floating-point literal '9.99999999999999999993E+38' overflows 32-bit floating-point type}}
```

Comment:
Can you get the textual representation of the literal so this would be reported as "floating-point literal '1E39' overflows 32-bit floating point type". It seems odd to call a value the programmer didn't write a 'literal'.